### DISCUSSION TITLE
Bluetooth key autoswitch between vehicles?

### DISCUSSION THREAD

**Discussion by LRRWD** [Kemmey/Kemmey-TeslaWatch-Public#569] [Ideas]
Hi! Love the Watch for Tesla app, even more so with the passive BT key feature.

A couple years ago, Tesla had made an update to their app that allows the phone as key feature to work with any car paired with phone without the need to open the app and set the vehicle as the primary. 

Is this a possibility in Watch OS?

**Comment by Kemmey** [comment]
It is absolutely technically possible - I've just not had the time to do it. It comes with a lot of complications for the internal design of the app, that basically just reveals years of poor planning on my part. I will have to do a major rewrite of some of the core mechanics of the app to make multi-vehicle support work as well as I want it - it is highly overdue, so one of these days...
